edbdeploy/data/terraform/aws/environments/ec2/outputs.tf

```diff
@@ -98,15 +146,204 @@ ssh-keygen -f ~/.ssh/known_hosts -R ${aws_instance.postgres_server[count].public
-%{endif ~}
+%{for barman_count  in range(var.barman_server["count"]) ~}
+ssh-keyscan -H ${aws_instance.barman_server[barman_count].public_ip} >> ~/.ssh/known_hosts
+ssh-keyscan -H ${aws_instance.barman_server[barman_count].public_ip} >> tpa_known_hosts
+ssh-keygen -f ~/.ssh/known_hosts -R ${aws_instance.barman_server[barman_count].public_dns}
+%{endfor ~}
 %{for count in range(var.pooler_server["count"]) ~}
 ssh-keyscan -H ${aws_instance.pooler_server[count].public_ip} >> ~/.ssh/known_hosts
+ssh-keyscan -H ${aws_instance.pooler_server[count].public_ip} >> tpa_known_hosts
 ssh-keygen -f ~/.ssh/known_hosts -R ${aws_instance.pooler_server[count].public_dns}
+%{endfor ~}
+%{for count in range(var.bdr_server["count"]) ~}
+ssh-keyscan -H ${aws_instance.bdr_server[count].public_ip} >> ~/.ssh/known_hosts
+ssh-keyscan -H ${aws_instance.bdr_server[count].public_ip} >> tpa_known_hosts
+ssh-keygen -f ~/.ssh/known_hosts -R ${aws_instance.bdr_server[count].public_dns}
+%{endfor ~}
+%{for count in range(var.bdr_witness_server["count"]) ~}
+ssh-keyscan -H ${aws_instance.bdr_witness_server[count].public_ip} >> ~/.ssh/known_hosts
+ssh-keyscan -H ${aws_instance.bdr_witness_server[count].public_ip} >> tpa_known_hosts
+ssh-keygen -f ~/.ssh/known_hosts -R ${aws_instance.bdr_witness_server[count].public_dns}
 %{endfor ~}
     EOT
 }
+
+resource "local_file" "TPAexecYamlConfig" {
+  filename = "${abspath(path.root)}/${var.tpaexec_config_yaml_filename}"
+  content  = <<EOT
+%{if var.bdr_server["count"] > 0 ~}
+---
+architecture: BDR-Always-ON
+cluster_name: ${var.cluster_name}
+cluster_tags: {}
+
+cluster_vars:
+  bdr_database: bdrdb
+  bdr_node_group: bdrgroup
+  extra_postgres_extensions:
+  - pglogical
+  failover_manager: harp
+  postgres_data_dir: /pgdata/pg_data
+  postgres_initdb_opts:
+  - --waldir=/pgwal/pg_wal
+  postgres_coredump_filter: '0xff'
+  postgres_version: '13'
+  postgresql_flavour: epas
+  preferred_python_version: python3
+  repmgr_failover: manual
+  tpa_2q_repositories:
+  - products/bdr_enterprise_3_7-epas/release
+  - products/pglogical3_7/release
+  - products/harp/release
+  use_volatile_subscriptions: false
+  publications:
+  - type: bdr
+    database: bdrdb
+    replication_sets:
+    - name: bdrgroup
+      autoadd_tables: false
+      replicate_delete: false
+      replicate_insert: false
+      replicate_truncate: false
+      replicate_update: false
+    - name: bdrdatagroup
+      autoadd_tables: true
+      replicate_delete: true
+      replicate_insert: true
+      replicate_truncate: true
+      replicate_update: true
+
+ssh_key_file: ${var.ssh_priv_key}
+
+locations:
+- Name: BDRDC1
+- Name: BDRDC2
+- Name: BDRDC3
+
+instance_defaults:
+  platform: bare
+  vars:
+    ansible_user: ${var.ssh_user}
+
+instances:
+%{for bdr_count in range(var.bdr_server["count"]) ~}
+%{if var.pg_type == "EPAS" ~}
+- Name: epas${bdr_count + 1}
+%{else ~}
+- Name: pgsql${bdr_count + 1}
```

Comment:
As we know, we set the hostname based on the node name. If we use postgresN then it won't be easy to know whether we are connected to EPAS or Postgres node. I am open to ideas. Let me know if you think otherwise.